action.yml

```diff
@@ -45,19 +45,5 @@ inputs:
         required: false
         description: Path to the report.json file to compare against. This should be from the branch the PR is merging into.
 runs:
-    using: composite
-    steps:
-        - run: $GITHUB_ACTION_PATH/run.sh
-          shell: bash
-          env:
-              INPUT_GITHUB-TOKEN: ${{ inputs.github-token }}
-              INPUT_TEST-SCRIPT: ${{ inputs.test-script }}
-              INPUT_THRESHOLD: ${{ inputs.threshold }}
-              INPUT_WORKING-DIRECTORY: ${{ inputs.working-directory }}
-              INPUT_ICONS: ${{ inputs.icons }}
-              INPUT_ANNOTATIONS: ${{ inputs.annotations }}
-              INPUT_PACKAGE-MANAGER: ${{ inputs.package-manager }}
-              INPUT_SKIP-STEP: ${{ inputs.skip-step }}
-              INPUT_CUSTOM-TITLE: ${{ inputs.custom-title }}
-              INPUT_COVERAGE-FILE: ${{ inputs.coverage-file }}
-              INPUT_BASE-COVERAGE-FILE: ${{ inputs.base-coverage-file }}
+    using: node12
+    main: 'dist/index.js'

```

Comment:
Need to use this to get some internal ENV Vars populated which is used in the @actions/cache package.

Reply:
The reason why the run.sh script was previously used is described here (#122). In short, the use of this script made it possible to not fix the version of the node - user can pick it manually. 
I've checked `@actions/cache` source code, so they are using only two environment variables - `GITHUB_EVENT_NAME` and `GITHUB_REF`. So, you can try to just pass these values, as it was in old configuration:
```yml
env:
    GITHUB_EVENT_NAME: ${{ env.GITHUB_EVENT_NAME }}
    GITHUB_REF: ${{ env.GITHUB_REF }}
```

Reply:
https://github.com/actions/toolkit/blob/ea81280a4d48fb0308d40f8f12ae00d117f8acb9/packages/cache/src/internal/cacheHttpClient.ts#L36

This is the env var that's required, and was not being populated for composite actions.

https://github.com/actions/toolkit/issues/852 

There seems to be an issue open in the repo.

Reply:
Does this work now? Are you sure the ENVs `ACTIONS_CACHE_URL` or `ACTIONS_RUNTIME_URL` are injected? I would suggest adding a console.log() to check it

Reply:
I logged process.env to check.
When running as composite action, even after injecting to env as @ArtiomTr suggested, I wasn't seeing the envs.
Only making it a node function worked. 

But tying it to node12 is a step back from current functionality.

Reply:
Working logs : https://github.com/sidharthv96/jest-coverage-report-action-test/pull/2/checks

Reply:
@sidharthv96 I need time to dig deeper into the problem and maybe find a solution. I hope you don't mind if this PR is delayed for a little.

Reply:
No problem at all, this was just one feature I thought might help speed up the test runs when writing the comments PR.

Nice to have, not need to have. Especially if it breaks compatability.

Reply:
One extremely hacky way we could enable someone who really needs to speed up tests is.

Something in documentation as an example.

I really don't think it's a good idea, but just throwing it out there 😂

- sh command to get base ref hash if it exists
- Use GitHub cache action with report path and ref as key
- Run our action as composite itself
- check if report exists, use it if does as base branch report.
- switch to head, run tests, compare.
- write **base report** back to path 
- GH cache action will store it. 

Not really sure if it's worth the effort though.